### PR TITLE
firmware: add target to flash radio_dongle

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/firmware/tools/flash_dongle.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  echo "INDEX_FOR_bin_FILE is chosen from:"
+  for i in "${!bin_files[@]}"; do
+    printf "%s)  %s\n" "$i" "${bin_files[$i]}"
+  done
+  exit 1
+fi
+bin_file=${bin_files[$1]}
+
+# Try to flash the given binary
+dfu-util -d 0483 -a "@Internal Flash  /0x08000000/04*016Kg,01*064Kg,07*128Kg" -D $bin_file  -s 0x08000000:leave
```
Comment (garethellis0) [inline src/firmware/tools/flash_dongle.sh]: Is this effectively the same as the script for flashing the (old) robots? Could we merge them?

**Review by garethellis0 — CHANGES_REQUESTED** [review]
Basically LGTM, just one comment. Also let's add a slightly more descriptive PR title so that this looks like a sane commit when it squashes to master.

File: src/firmware/tools/BUILD [hunk 2]
```diff
@@ -7,3 +7,13 @@ sh_binary(
     restricted_to = ["//cc_toolchain:stm32f4"],
     deps = ["@bazel_tools//tools/bash/runfiles"],
 )
+
+sh_binary(
+    name = "flash_dongle",
```
Comment (MathewMacDougall) [inline src/firmware/tools/BUILD]: Maybe lets use a slightly more desciptive name like "flash_radio_dongle"?

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
nothing much to add

File: src/firmware/tools/flash_dongle.sh (hunk 1 above)
Comment (akhilveeraghanta) [reply src/firmware/tools/flash_dongle.sh]: merged

File: src/firmware/tools/BUILD (hunk 2 above)
Comment (akhilveeraghanta) [reply src/firmware/tools/BUILD]: fixed

**Comment by MathewMacDougall** [comment]
Now that we've consolidated to a single `flash_firmware` target, does this change the exact command needed to flash the radio or robot? If it does, can you please update the README documentation @akhilveeraghanta ?

**Comment by akhilveeraghanta** [comment]
@MathewMacDougall there was no documentation on how to flash the old robots, just added a section and mentioned it there

File: src/firmware/tools/flash_dongle.sh (hunk 1 above)
Comment (garethellis0) [reply src/firmware/tools/flash_dongle.sh]: Nice, this is very clean now.

**Comment by garethellis0** [comment]
It's worth noting that if we don't pass any args to `flash_firmware`, it complains about an unbound variable. Not blocking, but we might want to add a quick check for that.

**Comment by MathewMacDougall** [comment]
Confirmed this works to flash both robots and dongles

**Comment by akhilveeraghanta** [comment]
@garethellis0 updated to fix that

**Review by garethellis0 — APPROVED** [review]

**Review by MathewMacDougall — APPROVED** [review]

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Pull request merged**